### PR TITLE
Add option for a Solution roll-up of BDIO Reports

### PR DESCRIPTION
**Pull request by romeara**
Allow users to specify flag to, in additional to individual BDIO files, generate a single BDIO which merges the project reports

### CODE REVIEW

**Comment by yevster** [comment]
What's the use case where this rollup flag might be false?

**Comment by romeara** [comment]
@akamen let me know you guys are a bit buried, and I'd love to get this squared away before CoPilot's "real" launch, so I thought I'd take a crack at it.

A disclaimer - I've never:
- Used Visual Studio before
- Written in C# before
So I apologize in advance for anything terrible; please feel free to point stuff out, I'll be more than happy to correct it.

As far as I can tell, this does in fact compile. However, I didn't see any tests in the repository, and VS didn't (seem to?) run any. As a result, I'd recommend:
- Someone more familiar than me trying this on a live project
- Running the branch through the integration tests
Before even considering merging... cause I don't want to accidentally break anything.

I did my best to follow what seemed to be the naming/setup conventions, though I expect we'll want to modify a few things before it's ready

**Comment by romeara** [comment]
Well that's ironic - CoPilot seems to have trouble with it's Hub right now and failed the build.... I'll poke IT

File: BuildBom/SolutionGenerator.cs [hunk 1]
```diff
@@ -77,6 +84,22 @@ public override bool Execute()
                         bool projectResult = base.Execute();
                         PackagesConfigPath = ""; // reset to use the project packages file.
                         result = result && projectResult;
+
+                        if (projectResult && GenerateMergedBdio)
+                        {
+                            string bdioFilePath = $"{OutputDirectory}{Path.DirectorySeparatorChar}{HubProjectName}.jsonld";
```
Comment (yevster) [inline BuildBom/SolutionGenerator.cs]: Use Path.combine. https://msdn.microsoft.com/en-us/library/fyy7a5kt(v=vs.110).aspx

It handles special cases that concatenation does not.

File: BuildBom/SolutionGenerator.cs [hunk 2]
```diff
@@ -137,5 +166,27 @@ public override bool Execute()
 
             return projectDataMap;
         }
+
+        private void GenerateMergedFile(List<BdioNode> components)
```
Comment (yevster) [inline BuildBom/SolutionGenerator.cs]: The C# convention is that public methods and properties start with an uppercase letter. Private methods and properties do not.

File: BuildBom/SolutionGenerator.cs (hunk 1 above)
Comment (romeara) [inline BuildBom/SolutionGenerator.cs]: Ok, shall use the library handling

**Comment by romeara** [comment]
@psantos1113 Looking through this again, my open concern is whether the other properties besides the output directory are bgin reset to the solution's values after generating the project BDIOs - let me load up VS 2015 on my mac and check, then we can try running the integration tests?

**Comment by romeara** [comment]
@psantos1113 I've added the previously mentioned restoration of project/version name on the solution level

**Comment by psantos1113** [comment]
Tested the code.  Added a few lines to set default projectName and version information.  Other than that the changes are good.

**Pull request merged**